Sources/ScipioKit/Producer/FrameworkProducer.swift

```diff
@@ -124,22 +124,26 @@ struct FrameworkProducer {
             }
         }
 
-        for target in targetsToBuild {
-            try await buildXCFrameworks(
-                target,
-                outputDir: outputDir,
-                buildOptionsMatrix: buildOptionsMatrix
-            )
-        }
+        let targetBuildResult = await buildTargets(targetsToBuild)
+
+        let builtTargets: OrderedSet<CacheSystem.CacheTarget> = switch targetBuildResult {
+            case .completed(let builtTargets),
+                 .interrupted(let builtTargets, _):
+                builtTargets
+            }
 
-        await cacheFrameworksIfNeeded(Set(targetsToBuild), cacheSystem: cacheSystem)
+        await cacheFrameworksIfNeeded(Set(builtTargets), cacheSystem: cacheSystem)
 
         if shouldGenerateVersionFile {
             // Versionfiles should be generate for all targets
-            for target in allTargets {
+            for target in builtTargets {
                 await generateVersionFile(for: target, using: cacheSystem)
             }
         }
+
+        if case .interrupted(_, error) = targetBuildResult {
```

Comment:
```suggestion
        if case .interrupted(_, let error) = targetBuildResult {
```

I'm sorry for my previous missuggestion. 🙇